lib/registry.js

```diff
@@ -16,7 +17,9 @@ export function define(id, spec) {
   if (get(id)) {
     console.warn(`Overwriting spec ${getName(id)}`)
   }
-  spec.name = getName(id)
+  if (!spec.name && typeof spec !== "symbol") {
```

Comment:
So an interesting finding is that Symbol.for() can take any kind of data, not just string. One could use a number or another object and make a symbol out of it. Do you want to restrict registry keys to Symbol and string?